src/dc_receive_imf.rs

```diff
@@ -1262,7 +1230,7 @@ async fn save_locations(
                         location::save(context, chat_id, from_id, locations, false)
                             .await
                             .unwrap_or_default();
-                    if newest_location_id != 0 && !hidden && !location_id_written {
```

Comment:
i have no idea why we checked for `!hidden` here before, so, not sure if that can be removed. maybe :) otoh, locations are experimental anyway :)

Reply:
All messages that were `hidden` here are now in the trash chat, and `save_locations` quits early when `ChatId` is special, so there should be no behavior change.

Reply:
ah, now i understand. the `!hidden` was just some optimization to not save location to the dummy messages.